src/include/homestore/btree/detail/btree_node.hpp

```diff
@@ -619,8 +619,10 @@ class BtreeNode : public sisl::ObjLifeCounter< BtreeNode > {
 
     friend void intrusive_ptr_release(BtreeNode* node) {
         if (node->m_refcount.decrement_testz(1)) {
+            auto node_buffer = node->m_phys_node_buf;
             node->~BtreeNode();
             delete[] uintptr_cast(node);
+            delete[] node_buffer;
```

Comment:
It seems odd to deallocate in a different order than allocation. Originally the BtreeNode is allocated, then the node_buf is allocated, constructed and finally the BtreeNode is constructed. Should we not to the same in reverse? :
```
            auto node_buffer = node->m_phys_node_buf;
            node->~BtreeNode();
            delete[] node_buffer;
            delete[] uintptr_cast(node);
```

Reply:
Also, it's not clear that `node_buffer` was allocated with `new[]` anywhere. We should define a specific type for this so someone cannot allocate with `new` or `malloc` and expect this class to work right?

Reply:
The sequence is 
create_root_node() -> alloc_leaf_node() -> alloc_node() [[node_buffer is created here]](https://raw.githubusercontent.com/eBay/HomeStore/fiber/src/include/homestore/btree/mem_btree.hpp#:~:text=BtreeNodePtr%20alloc_node(bool%20is_leaf)%20override%20%7B), then new_node is created and the allocated [node_buffer is passed](https://raw.githubusercontent.com/eBay/HomeStore/fiber/src/include/homestore/btree/detail/btree_node_mgr.ipp#:~:text=static%20BtreeNode*%20create_node(uint32_t%20node_ctx_size%2C%20Args%26%26...%20args)%20%7B) .